components/supervisor/pkg/ports/ports_test.go

```diff
@@ -443,6 +443,56 @@ func TestPortsUpdateState(t *testing.T) {
 				{{LocalPort: 5900, Served: true, Exposed: &api.ExposedPortInfo{Visibility: api.PortVisibility_private, OnExposed: api.OnPortExposedAction_notify_private, Url: "foobar"}}},
 			},
 		},
+		{
+			Desc: "port status has description set as soon as the port gets exposed, if there was a description configured",
+			Changes: []Change{
+				{
+					Config: &ConfigChange{workspace: []*gitpod.PortConfig{
+						{Port: 8080, Visibility: "private", Description: "Development server"},
+					}},
+				},
+				{
+					Served: []ServedPort{{"00000000", 8080, false}},
+				},
+				{
+					Exposed: []ExposedPort{{LocalPort: 8080, Public: false, URL: "foobar"}},
+				},
+			},
+			ExpectedExposure: []ExposedPort{
+				{LocalPort: 8080},
+			},
+			ExpectedUpdates: UpdateExpectation{
+				{},
+				{{LocalPort: 8080, Description: "Development server"}},
+				{{LocalPort: 8080, Description: "Development server", Served: true}},
+				{{LocalPort: 8080, Description: "Development server", Served: true, Exposed: &api.ExposedPortInfo{Visibility: api.PortVisibility_private, OnExposed: api.OnPortExposedAction_notify, Url: "foobar"}}},
+			},
+		},
+		{
+			Desc: "ports with no description configured will remain with an empty description in their status",
```

Comment:
Is not is already covered by existing tests?

Reply:
True, it's essentially the same as the test from lines 239-263.
I'm removing it.